images/push-stages.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+CYAN='\033[0;36m'
+NC='\033[0m'
+
+repository="quay.io/pusher"
+
+for arg in "$@"; do
+  case ${arg%%=*} in
+   "--build-root")
+      build_root="${arg##*=}"
+      ;;
+    "--help")
+      printf "${GREEN}$0${NC}\n"
+      printf "  available options:\n"
+      printf "  --build-root=${BLUE}<docker build root>${NC}\n"
+      exit 0
+      ;;
+    *)
+      echo "Unknown option: $arg"
+      exit 2
+      ;;
+    esac
+done
+
+
+###########################################
+#
+# Check if the image is a multi stage build
+#
+###########################################
+lines=$(cat ${build_root}/Dockerfile | grep 'FROM')
```

Comment:
```suggestion
lines=$(grep 'FROM .* AS' ${build_root}/Dockerfile)
```
And as above you won't need the if statement in the while loop.